src/HttpHeader.cc

```diff
@@ -1114,6 +1111,43 @@ HttpHeader::putExt(const char *name, const char *value)
     addEntry(new HttpHeaderEntry(Http::HdrType::OTHER, SBuf(name), value));
 }
 
+void
+HttpHeader::updateOrAddStr(const Http::HdrType id, const SBuf &newValue)
+{
+    assert(any_registered_header(id));
+    assert(Http::HeaderLookupTable.lookup(id).type == Http::HdrFieldType::ftStr);
+
+    // XXX: HttpHeaderEntry::value suffers from String size limits
+    Must(newValue.length() < String::SizeMaxXXX()); // TODO: Assure() in master/v6
```

Comment:
Now this is against master the TODO can be implemented.

Reply:
@wweric, I cannot implement this TODO because this PR is using a branch that is protected in your repository (or, at the very least, that is how I interpret what GitHub tells me when I try to edit this file). Please 
* replace this specific Must() with Assure();
* `#include "base/Assure.h"` (immediately after "squid.h") at the top of this .cc file; and
* remove the TODO comment on this line. The XXX comment above it should stay.

P.S. FWIW, it is best to use dedicated, PR-specific branches for PRs, but there is no need to adjust this PR branch.

Reply:
@wweric, just to avoid misunderstanding: AFAICT, this is the last change request that should be addressed before this PR is merged.